install/kubernetes/ingress-sni-zvpn.yaml

```diff
@@ -0,0 +1,335 @@
+                      cluster_name: [rds]
+                # envoy.router factory returned nullptr instead of empty config message.
+                #              http_filters:
+                #              - name: envoy.router
+                #                deprecated_v1: true
+
+      # HTTPS - currently manually configured, until pilot generates LDSv2 for ingress.
+      # If using kube-lego, certificate can be retrieved for testing with:
+      #   kubectl get secret -n istio-system httpbin-tls -o jsonpath='{.data.tls\.key}' |base64 -d > /etc/istio/httpbin-certs/tls.key
+      #   kubectl get secret -n istio-system httpbin-tls -o jsonpath='{.data.crt\.key}' |base64 -d > /etc/istio/httpbin-certs/crt.key
+      - name: http443
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 8443 }
+        filter_chains:
+        - filter_chain_match:
+            sni_domains: "httpbin.weekly.istio.webinf.info"
+          tls_context:
+            common_tls_context:
+              tls_certificates:
+                - certificate_chain: { filename: "/etc/istio/httpbin-certs/tls.crt" }
+                  private_key: { filename: "/etc/istio/httpbin-certs/tls.key" }
+          filters:
+          - name: envoy.http_connection_manager
+            config:
+              stat_prefix: http443
+              codec_type: AUTO
+              rds:
+                route_config_name: "443"
+                config_source:
+                  api_config_source:
+                      api_type: REST_LEGACY
+                      refresh_delay: 10s
+                      cluster_name: [rds]
+
+      # Example for testing explicit routes (route_config vs rds)
+      - name: http8081
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 8081 }
+        filter_chains:
+        - filters:
+          - name: envoy.http_connection_manager
+            # TODO: add mixer
+            config:
+              stat_prefix: ingress_http8080
+              codec_type: AUTO
+              route_config:
+                name: local_route
+                virtual_hosts:
+                - name: local_service
+                  domains: ["*"]
+                  routes:
+                  - match: { prefix: "/" }
+                    route: { cluster: out.pilot }
+
+
+      - name: tcp-zvpn-pilot
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 15003 }
+        filter_chains:
+        - filters:
+          - name: envoy.tcp_proxy
+            config:
+              # listener_manager_impl.cc - if false, will attempt to convert to proto - not implemented for tcp_proxy
+              deprecated_v1: true
+              value:
+                stat_prefix: tcp_pilot
+                route_config:
+                    routes:
+                    - cluster: zvpn.pilot
+
+      - name: tcp-zvpn-ca
+        address:
+          socket_address: { address: 0.0.0.0, port_value: 8060 }
```

Comment:
why expose CA

Reply:
or pilot, etc. ?

Reply:
That's part of the 'zero vpn' - machines without VPN/VPC in other clusters (or raw VMs) should be able to access pilot and CA (if they have bootstrap certs). 

It's a plain TCP proxy, with TLS terminated in CA and pilot.

Reply:
To be clear - my primary goal with the PR is to get a working zero-vpn ( mesh expansion without requiring the internal load balancer and VPC/VPN). The SNI is a secondary goal, doing them at the same time since both require v2

Reply:
So is this some prototype or real implementation?

Reply:
It's the first step towards a real implementation :-) It should be usable and work - but there are many changes needed to improve the UX and simplify. Step one is getting SNI and zvpn using 
a manually configuration, step 2 is to automate..